common/src/main/java/com/revenuecat/purchases/common/networking/ETagManager.kt

```diff
@@ -0,0 +1,120 @@
+package com.revenuecat.purchases.common.networking
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.revenuecat.purchases.common.sha1
+import org.json.JSONObject
+import java.net.HttpURLConnection
+
+private const val NOT_MODIFIED_RESPONSE_CODE = 304
+private const val INTERNAL_SERVER_ERROR_RESPONSE_CODE = 500
```

Comment:
should these live somewhere shared? I feel like this won't be the only usage of `500` in the codebase

Reply:
Yes good call. I noticed iOS has a HTTPStatusCodes enum we should replicate